deltacat/utils/pyarrow.py

```diff
@@ -33,15 +33,74 @@
 from deltacat.utils.common import ContentTypeKwargsProvider, ReadKwargsProvider
 from deltacat.utils.performance import timed_invocation
 from deltacat.utils.daft import daft_s3_file_to_table
-from deltacat.utils.arguments import sanitize_kwargs_to_callable
+from deltacat.utils.arguments import (
+    sanitize_kwargs_to_callable,
+    sanitize_kwargs_by_supported_kwargs,
+)
 
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
+
+def _filter_schema_for_columns(schema: pa.Schema, columns: List[str]) -> pa.Schema:
+
+    target_schema_fields = []
+
+    for column_name in columns:
+        index = schema.get_field_index(column_name)
+
+        if index != -1:
+            target_field = schema.field(index)
+            target_schema_fields.append(target_field)
+
+    target_schema = pa.schema(target_schema_fields, metadata=schema.metadata)
+
+    return target_schema
+
+
+def pyarrow_read_csv(*args, **kwargs) -> pa.Table:
+    try:
+        kwargs = sanitize_kwargs_by_supported_kwargs(
+            ["read_options", "parse_options", "convert_options", "memory_pool"], kwargs
+        )
+        return pacsv.read_csv(*args, **kwargs)
+    except pa.lib.ArrowInvalid as e:
+        if e.__str__() == "Empty CSV file":
```

Comment:
This is a workaround specific to quirks of our own internal data catalog, which ideally shouldn't be producing empty CSVs anyway. As a general solution, perhaps its better to provide an option to disable/enable this functionality if desired?

Reply:
Good point.